### ISSUE TITLE
--export-key does not work

### ISSUE THREAD

**Issue by ghost** [neosmart/SecureStore#10]
Hello,

I recently started to implement Secure Store in my .Net Backendservice. I saw that there is the "--export-key" command to export a key after the secure store was already created. That would be exactly what I am looking for, but I can't seem to make it work.
I use the .net tool for reference, not the one from rust.
Is it possible that that feature is not yet released? Because I saw on [nuget.org](https://www.nuget.org/packages/SecureStore.Client/1.1.1#versions-body-tab) that the newest Version 1.1.1 was last updated 5 months ago, while the "--export-key" feature commit is two months old: 
https://github.com/neosmart/SecureStore/commits/master/Client
If I got all that correct, is it possible to release the newest version? It would help me out a ton.

Thank you very much in advance!

**Comment by mqudsi** [comment]
Hello @Faeber and thanks for being a SecureStore user. I am pushing it to nuget now but unfortunately it will take maybe another half hour or so because I have to first publish the `SecureStore` package before I can publish the `SecureStore.Client` package (because it depends on it and nuget has a delay).

**Comment by mqudsi** [comment]
The current code has been pushed to nuget as a preview release, install with 

```
dotnet tool install --global SecureStore.Client --version 1.2.0-preview1
```

or if it's already installed:

```
dotnet tool update --global SecureStore.Client --version 1.2.0-preview1
```

Note that the exported key will be in the new PEM-like format (copy-and-paste friendly, ascii instead of binary) so you will need to upgrade your project to use the same version of `SecureStore` (`1.2.0-preview1`) to use the resulting key.

**Comment by ghost** [comment]
Hi @mqudsi 
Thank you very much for the quick answers! I got the newest version, but it does not work as I expected:
![grafik](https://user-images.githubusercontent.com/120490468/207530214-e51b9359-0186-4ddc-ada8-5b2fb8649038.png)
First, I create a SecureStore with a password and then I want to export a key without overwriting the content inside the SecureStore. No key is being exported. The content can't be overwritten because the store could already contain secrets in my case.
It does work however, if I overwrite the existing store:
![grafik](https://user-images.githubusercontent.com/120490468/207531785-ca09d753-10a9-4475-9e58-fab11cbabf44.png)
Is this the intended behavior? If so, it would not be much of a problem. The new copy-paste-friendly keyfile provides an easy workaround for me. I just want to verify that I didn't stumble across a bug.

Additional information:
It's the same if I use the commands in a .net project. The latest version should also be installed correctly although, the -h option generates an error:
![grafik](https://user-images.githubusercontent.com/120490468/207534222-1e8dc8e5-6ea4-4fc4-a191-490d756438f7.png)

**Comment by OneCyrus** [comment]
Just hit the same issue. I created a store with a password and would like to also generate a key-file for this store. The create command overwrites my store and everything is lost in there.

@mqudsi: is there a way to export a key-file AFTER a store with a password is created?

**Comment by mqudsi** [comment]
The new `--export-key` option can be used at any time with any subcommand to export a copy of the key.

@Faeber Using `SecureStore create foo.json` then `SecureStore create foo.json --export-key foo.key` overwrites the old store the second time - and if you say `n` to overwriting the old store, the key isn't exported. You don't want to do that. Instead use `--export-key` with any other command, such as `SecureStore get -a --export-key foo.key`.

I am pushing a new preview release to Nuget with the fix for the exception when running `SecureStore -h`.

**Comment by mqudsi** [comment]
I did find the issue where `--export-key` was only working in conjunction with `SecureStore create` and that's been fixed.

**Comment by mqudsi** [comment]
The latest release is available on nuget as 1.2.0-preview2
